### PR TITLE
fix: cve resolution for 74-5 and consistency for internal subproject …

### PR DESCRIPTION
**Pull request by Tallicia**
…build dependencies. 

Directing subproject gradle builds to use the latest working versions of dependency libraries with identified CVEs.

### CODE REVIEW

**Comment by Tallicia** [comment]
@peterhaochen47 Thank you for the feedback! :) Refactor is included now.

File: dependencies.gradle [hunk 1]
```diff
@@ -16,6 +18,9 @@ versions.springSecuritySamlVersion = "1.0.10.RELEASE" // this lib has reached EO
 versions.tomcatVersion = "9.0.75"
 versions.statsdClientVersion = "3.1.0" // statsd is not included in spring-boot 2.2 and later, so we need to specify the dependency explicitly
 versions.junitVersion = "4.13.2" // Needed to support junit 4 with spring-boot 2.4
+versions.snakeyamlVersion = "1.33" // Needed to resolve CVEs in internal spring-boot 2.7.12 inclusion of snakeyaml
```
Comment (peterhaochen47) [inline dependencies.gradle]: Why does this (plus the usage of this var below) need to be defined in addition to `ext["snakeyaml.version"] = "1.33"`. I've tried to remove this var & did not notice any difference in our final dependencies (as shown in intellij) after removing.

Comment (peterhaochen47) [reply dependencies.gradle]: Other than ^, LGTM

Comment (Tallicia) [reply dependencies.gradle]: ~~It is redundant in the this cve only set of 4 libraries. I'll remove it.~~

Comment (Tallicia) [reply dependencies.gradle]: Correction, removing it results in a build failure: 
`/Users/yalicia/src/workspace/uaa/server/src/main/java/org/cloudfoundry/identity/uaa/impl/config/UaaConfiguration.java:21: error: package org.yaml.snakeyaml does not exist
import org.yaml.snakeyaml.TypeDescription;
                         ^`

Comment (peterhaochen47) [reply dependencies.gradle:21]: I've tried removing this var `versions.snakeyamlVersion = "1.33"` and its usage (aka changing `libraries.snakeyaml = "org.yaml:snakeyaml:${versions.snakeyamlVersion}"` back to libraries.snakeyaml = "org.yaml:snakeyaml" which is the existing code) and it seems to be work. Did you experiment with the same thing?

Comment (peterhaochen47) [reply dependencies.gradle]: I've tried removing this var `versions.snakeyamlVersion = "1.33"` and its usage (aka changing `libraries.snakeyaml = "org.yaml:snakeyaml:${versions.snakeyamlVersion}"` back to `libraries.snakeyaml = "org.yaml:snakeyaml"` which is the existing code) and it seems to be work (with `./gradlew build -x test`) and does not result in warnings in "Dependency Analyzer" tool in intellij. Did you experiment with the same thing?

Comment (Tallicia) [reply dependencies.gradle]: The redundant snakeyaml versioning has been removed.

**Review by peterhaochen47 — APPROVED** [review]

**Pull request merged**